### PR TITLE
image-rs: add feature gate for verity

### LINKED ISSUE #314: image-rs: add feature gate to verify
it's pulled in unconditionally and expects host headers to be installed. Can it be behind a feature gate?

### PR DESCRIPTION
**Pull request by ChengyuZhu6**
add feature gate for verity. 

Fixes #314

### CODE REVIEW

**Comment by ChengyuZhu6** [comment]
cc @mythi

File: image-rs/src/verity/dmverity.rs [hunk 1]
```diff
@@ -99,6 +99,16 @@ impl DmVerityOption {
             hash: parts[2].to_string(),
         })
     }
+    /// Serialize the virtual volume object to json.
```
[the remark this answers is not in the document]
Comment (ChengyuZhu6) [reply image-rs/src/verity/dmverity.rs]: Fixed.

**Comment by mythi** [comment]
@ChengyuZhu6 thanks a lot! This looks good but I was wondering that with this change, can we also clean-up some of the CI steps. For example, [9dc732a](https://github.com/confidential-containers/guest-components/commit/9dc732ab06a23fa3830a342fae67f83f218c41ad) can probably be reverted?

**Comment by ChengyuZhu6** [comment]
> @ChengyuZhu6 thanks a lot! This looks good but I was wondering that with this change, can we also clean-up some of the CI steps. For example, [9dc732a](https://github.com/confidential-containers/guest-components/commit/9dc732ab06a23fa3830a342fae67f83f218c41ad) can probably be reverted?

hi @mythi . One question:  Does the 'devmapper' cause any blocking issues?

**Comment by mythi** [comment]
> > @ChengyuZhu6 thanks a lot! This looks good but I was wondering that with this change, can we also clean-up some of the CI steps. For example, [9dc732a](https://github.com/confidential-containers/guest-components/commit/9dc732ab06a23fa3830a342fae67f83f218c41ad) can probably be reverted?
> 
> hi @mythi . One question: Does the 'devmapper' cause any blocking issues?

It's not blocking anything but it's useless for enclave-cc for example.

**Review by jiangliu — APPROVED** [review]

**Pull request merged**